site/content/integrate/apps/deployment/_index.md

```diff
@@ -0,0 +1,106 @@
+
+A language library is used to emulate an HTTP to your app. For go you might use https://github.com/awslabs/aws-lambda-go-api-proxy.
+
+Finally you need to define the AWS function in the manifest of your app by adding a `aws_lambda` to it which has the following fields:
+
+
+| Name      | Description                                                                                                  |
+| :-------- | :----------------------------------------------------------------------------------------------------------- |
+| `path`    | The lambda function with its Path the longest-matching prefix of the call's Path will be invoked for a call. |
+| `name`    | A human readable name.                                                                                       |
+| `handler` | The name of the handler function.                                                                            |
+| `runtime` | The AWS lambda runtime to use.                                                                               |
+
+For a go app the manifest snippet would look like this:
+```json
+{
+    "aws_lambda": [
+        {
+            "path": "/",
+            "name": "go-function",
+            "handler": "$YOUR_APP_NAME",
+            "runtime": "go1.x"
+        }
+    ]
+}
+``` 
+
+### Deploying third-party apps to AWS
+
+Note that third-party apps are not supported in the Mattermost Cloud, they are used only for developer testing convenience. Provisioning in the third party AWS cloud environment is done by the **appsctl** tool using the command:
+
+```bash
+go install github.com/mattermost/mattermost-plugin-apps/cmd/appsctl@latest
+appsctl provision app /PATH/TO/YOUR/APP/BUNDLE
+```
+
+It reads appropriate AWS credentials from environment variables:
+
+`APPS_PROVISION_AWS_ACCESS_KEY`
+
+`APPS_PROVISION_AWS_SECRET_KEY`
+
+We need an app bundle to provision an app. The bundle might be provisioned from the local disk, from S3 (not implemented yet) or from some URL (not implemented yet). Provisioning consists of three parts:
+
+1. Creating the lambda functions with appropriate policies.
+2. Storing static assets in the dedicated S3 bucket.
+3. Storing app’s manifest file in the same dedicated S3 bucket.
+
+AWS Lambda functions have semantic names, which means that a function described in the manifest.json file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example a name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. App developer does not need to worry about the AWS lambda function names, Apps Plugin takes care of it. Dedicated S3 bucket name is stored in the environment variable:
```

Comment:
```suggestion
AWS Lambda functions have semantic names, which means that a function described in the `manifest.json` file translates to AWS as `$appID_$appVersion_$functionName` to avoid collisions with other apps' or other versions' functions. And **appsctl** provisions lambda functions using this name. For example a name of a `servicenow` app's lambda function might be `com-mattermost-servicenow_0-1-0_go-function`. You don't need to worry about the AWS Lambda function names - the Apps Plugin takes care of it. The dedicated S3 bucket name is stored in the environment variable:
```